File: src/main/java/com/paragon/client/systems/module/hud/impl/Armour.java
package com.paragon.client.systems.module.hud.impl;

import com.paragon.api.util.render.RenderUtil;
import com.paragon.api.util.world.BlockUtil;
import com.paragon.client.systems.module.hud.HUDModule;
import com.paragon.client.systems.module.setting.Setting;
import net.minecraft.client.Minecraft;
import net.minecraft.init.Blocks;
import net.minecraft.item.ItemStack;
import org.lwjgl.opengl.GL11;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Armour extends HUDModule {

    public static Armour INSTANCE;

    public static Setting<Boolean> waterOffset = new Setting<>("Water Offset", true)
            .setDescription("Position higher when you are underwater");

    public Armour() {
        super("Armour", "Displays your armour on screen");

        INSTANCE = this;
    }


    @Override
    public void render() {
        List<ItemStack> armourList = new ArrayList<>(mc.player.inventory.armorInventory);
        Collections.reverse(armourList);

        GL11.glPushMatrix();
        GL11.glTranslatef(0, waterOffset.getValue() &&
                // We do this rather than mc.player.isInWater() because we only want to offset if we can see the bubbles (top and bottom half of player is in water)
                // This checks that the top half of the player is in water
                BlockUtil.getBlockAtPos(mc.player.getPosition().up()).equals(Blocks.WATER) ? -10 : 0, 0);

        float xSpacing = 0;
        for (ItemStack itemStack : armourList) {
            // We don't want to render stack
            if (itemStack.isEmpty()) {
                xSpacing += 18;
                continue;
            }

            // Render stack
            RenderUtil.renderItemStack(itemStack, (getX() + xSpacing), getY() + 4, true);

            // Get the item's damage percentage
            int itemDamage = (int) (100 - ((1 - (((float) itemStack.getMaxDamage() - (float) itemStack.getItemDamage()) / (float) itemStack.getMaxDamage())) * 100));

            // Scale
            GL11.glScalef(0.75f, 0.75f, 0.75f);
            float scaleFactor = 1 / 0.75f;

            // Render the damage percentage
            Minecraft.getMinecraft().fontRenderer.drawStringWithShadow(String.valueOf(itemDamage), (getX() + xSpacing + (9 - (getStringWidth(String.valueOf(itemDamage)) / 2f))) * scaleFactor, getY() * scaleFactor, -1);

            GL11.glScalef(scaleFactor, scaleFactor, scaleFactor);

            xSpacing += 18;
        }

        GL11.glPopMatrix();
    }

    @Override
    public float getWidth() {
        return 18 * 4;
    }

    @Override
    public float getHeight() {
        return 22;
    }
}

File: src/main/kotlin/com/paragon/client/systems/module/hud/impl/Armour.kt
package com.paragon.client.systems.module.hud.impl

import com.paragon.api.setting.Setting
import com.paragon.api.util.render.RenderUtil.renderItemStack
import com.paragon.api.util.world.BlockUtil.getBlockAtPos
import com.paragon.client.systems.module.hud.HUDModule
import net.minecraft.client.Minecraft
import net.minecraft.init.Blocks
import net.minecraft.item.ItemStack
import org.lwjgl.opengl.GL11.*

object Armour : HUDModule("Armour", "Displays your armour on screen") {

    private val waterOffset = Setting("Water Offset", true, null, null, null).describedBy("Position higher when you are underwater")

    override fun render() {
        val armourList: ArrayList<ItemStack> = ArrayList<ItemStack>(minecraft.player.inventory.armorInventory)

        armourList.reverse()

        glPushMatrix()
        glTranslatef(0f, (if (waterOffset.value!! && minecraft.player.position.up().getBlockAtPos() == Blocks.WATER) -10 else 0).toFloat(), 0f)

        var xSpacing = 0f

        for (itemStack in armourList) {
            // We don't want to render stack
            if (itemStack.isEmpty) {
                xSpacing += 18f
                continue
            }

            // Render stack
            renderItemStack(itemStack, x + xSpacing, y + 4, true)

            // Get the item's damage percentage
            val itemDamage = (100 - (1 - (itemStack.maxDamage.toFloat() - itemStack.itemDamage.toFloat())) / itemStack.maxDamage.toFloat() * 100).toInt()

            // Scale
            glScalef(0.75f, 0.75f, 0.75f)
            val scaleFactor = 1 / 0.75f

            // Render the damage percentage
            Minecraft.getMinecraft().fontRenderer.drawStringWithShadow(itemDamage.toString(), (x + xSpacing + 9 - minecraft.fontRenderer.getStringWidth(itemDamage.toString()) / 2) * scaleFactor, y * scaleFactor, -1)
            glScalef(scaleFactor, scaleFactor, scaleFactor)
            xSpacing += 18f
        }

        glPopMatrix()
    }

    override var width: Float = 18f * 4f

    override var height: Float = 22f

}